src/client_builder.rs

```diff
@@ -116,19 +237,23 @@ where
     /// # fn main() -> Result<(), imap::Error> {
     /// let client = ClientBuilder::new("imap.example.com", 993)
     ///     .starttls()
-    ///     .connect(|domain, tcp| {
+    ///     .connect_tls_with(|domain, tcp| {
     ///         let ssl_conn = RustlsConnector::new_with_native_certs()?;
     ///         Ok(ssl_conn.connect(domain, tcp)?)
     ///     })?;
     /// # Ok(())
     /// # }
     /// ```
-    pub fn connect<F, C>(&mut self, handshake: F) -> Result<Client<C>>
+    pub fn connect_tls_with<F, C>(&mut self, handshake: F) -> Result<Client<C>>
```

Comment:
Note that this isn't _necessarily_ for TLS. This can be used for _any_ customization of the underlying I/O channel. I think it should just be called `connect_with`.